python/cuml/common/base.pyx

```diff
@@ -353,6 +355,83 @@ class Base(TagsMixin,
         else:
             self.n_features_in_ = X.shape[1]
 
+    def _get_tags(self):
+        # method and code based on scikit-learn 0.21 _get_tags functionality:
+        # https://scikit-learn.org/stable/developers/develop.html#estimator-tags
+        collected_tags = deepcopy(_default_tags)
+        for cl in reversed(inspect.getmro(self.__class__)):
+            if hasattr(cl, '_more_tags') and cl != Base:
+                more_tags = cl._more_tags(self)
+                collected_tags.update(more_tags)
+        return collected_tags
+
+
+class RegressorMixin:
+    """Mixin class for regression estimators in cuML"""
+
+    _estimator_type = "regressor"
+
+    @generate_docstring(
+        return_values={
+            'name': 'score',
+            'type': 'float',
+            'description': 'R^2 of self.predict(X) '
+                           'wrt. y.'
+        })
+    @cuml.internals.api_base_return_any_skipall
+    def score(self, X, y, **kwargs):
+        """
+        Scoring function for regression estimators
+
+        Returns the coefficient of determination R^2 of the prediction.
+
+        """
+        from cuml.metrics.regression import r2_score
+
+        if hasattr(self, 'handle'):
+            handle = self.handle
+        else:
+            handle = None
+
+        preds = self.predict(X, **kwargs)
+        return r2_score(y, preds, handle=handle)
+
+    def _more_tags(self):
+        return {
+            'requires_y': True
+        }
+
+
+class ClassifierMixin:
+    """Mixin class for classifier estimators in cuML"""
+
+    _estimator_type = "classifier"
+
+    @generate_docstring(
+        return_values={
+            'name':
+                'score',
+            'type':
+                'float',
+            'description': ('Accuracy of self.predict(X) wrt. y '
+                            '(fraction where y == pred_y)')
+        })
+    @cuml.internals.api_base_return_any_skipall
+    def score(self, X, y, **kwargs):
+        """
+        Scoring function for classifier estimators based on mean accuracy.
+
+        """
+        from cuml.metrics.accuracy import accuracy_score
+
+        if hasattr(self, 'handle'):
+            handle = self.handle
+        else:
+            handle = None
+
+        preds = self.predict(X, **kwargs)
+        return accuracy_score(y, preds, handle=handle)
+
```

Comment:
This code is now located in https://github.com/rapidsai/cuml/blob/branch-0.19/python/cuml/common/mixins.py, mentioning it so it is removed before merge